code/modules/atmospherics/machinery/trinary/mixer.dm

```diff
@@ -0,0 +1,187 @@
+
+#define _SET_SIGNAL_GAS(GAS, _, _, MIXTURE, ID, ...) signal.data[ID + #GAS] = round(100*##MIXTURE.GAS/MIXTURE##_total_moles);
+#define _RESET_SIGNAL_GAS(GAS, _, _, ID, ...) signal.data[ID + #GAS] = 0;
+#define SET_SIGNAL_MIXTURE(MIXTURE, ID) APPLY_TO_GASES(_SET_SIGNAL_GAS, MIXTURE, ID)
+#define RESET_SIGNAL_MIXTURE(ID) APPLY_TO_GASES(_RESET_SIGNAL_GAS, ID)
+
+/obj/machinery/atmospherics/trinary/mixer
+	name = "Gas mixer"
+	icon = 'icons/obj/atmospherics/mixer.dmi'
+	icon_state = "intact_off"
+	layer = PIPE_MACHINE_LAYER
+	plane = PLANE_NOSHADOW_BELOW
+
+	var/id_tag
+	var/master_id
```

Comment:
what are these id tags for btw

Reply:
Asking stuff and modifying what the mixer mixes.